modules/seqexec/server/src/main/scala/seqexec/server/keywords/DhsClientHttp.scala

```diff
@@ -8,113 +8,85 @@ import java.time.format.DateTimeFormatter
 import java.util.concurrent.atomic.AtomicInteger
 
 import org.log4s._
-import argonaut._
-import Argonaut._
-import cats.data.EitherT
 import cats.effect.IO
+import cats.effect.Timer
+import cats.data.EitherT
+import cats.implicits._
 import gem.enum.DhsKeywordName
 import seqexec.model.dhs.ImageFileId
-import seqexec.server._
+import seqexec.server.{SeqAction, SeqexecFailure, TrySeq}
 import seqexec.server.keywords.DhsClient.ImageParameters
 import seqexec.server.SeqexecFailure.SeqexecExceptionWhile
-import org.apache.commons.httpclient.HttpClient
-import org.apache.commons.httpclient.methods.{EntityEnclosingMethod, PostMethod, PutMethod}
-
-import scala.io.Source
+import io.circe.syntax._
 import cats.implicits._
+import io.circe.{Decoder, DecodingFailure, Encoder, Json}
+import org.http4s.client.Client
+import org.http4s._
+import org.http4s.dsl.io._
+import org.http4s.client.dsl.io._
+import org.http4s.circe._
+import org.http4s.client.middleware.{Retry, RetryPolicy}
+import scala.concurrent.duration._
 
 /**
   * Implementation of DhsClient that interfaces with the real DHS over the http interface
   */
-class DhsClientHttp(val baseURI: String) extends DhsClient {
+class DhsClientHttp(base: Client[IO], baseURI: Uri)(implicit timer: Timer[IO]) extends DhsClient {
   import DhsClientHttp._
 
-  // Connection timeout, im milliseconds
-  private val timeout = 10000
-
-  implicit def errorDecode: DecodeJson[Error] = DecodeJson[Error]( c => for {
-      t   <- (c --\ "type").as[ErrorType]
-      msg <- (c --\ "message").as[String]
-    } yield Error(t, msg)
-  )
-
-  implicit def obsIdDecode: DecodeJson[TrySeq[ImageFileId]] = DecodeJson[TrySeq[ImageFileId]](c => {
-    val r = c --\ "response"
-    val s = (r --\ "status").as[String]
-    s.flatMap {
-      case "success" =>
-        (r --\ "result").as[String].map(TrySeq(_))
-      case "error"   =>
-        (r --\ "errors").as[List[Error]].map(l =>
-          TrySeq.fail[ImageFileId](SeqexecFailure.Unexpected(l.mkString(", ")))
-        )
-      case r         =>
-        DecodeResult.fail(s"Unknown response: $r", s.history.getOrElse(CursorHistory.empty))
-    }
-  } )
-
-  implicit def unitDecode: DecodeJson[TrySeq[Unit]] = DecodeJson[TrySeq[Unit]]( c => {
-    val r = c --\ "response"
-    val s = (r --\ "status").as[String]
-    s flatMap {
-      case "success" =>
-        DecodeResult.ok(TrySeq(()))
-      case "error"   =>
-        (r --\ "errors").as[List[Error]].map(
-          l => TrySeq.fail[Unit](SeqexecFailure.Unexpected(l.mkString(", "))))
-      case r         =>
-        DecodeResult.fail(s"Unknown response: $r", s.history.getOrElse(CursorHistory.empty))
+  @SuppressWarnings(Array("org.wartremover.warts.Var"))
+  private val client = {
+    val max = 2
```

Comment:
Perhaps we should 3 or even more attempts. We really don't want to stop a sequence if there is an intermittent issue on the DHS, OTOH we don't want to wait too long either